crates/sui-move/src/build.rs

```diff
@@ -96,3 +108,42 @@ pub fn resolve_lock_file_path(
     build_config.lock_file = Some(lock_file_path);
     Ok(build_config)
 }
+
+pub const PUBLISHED_AT_MANIFEST_FIELD: &str = "published-at";
+
+pub struct SuiPackageHooks {}
+
+impl PackageHooks for SuiPackageHooks {
+    fn custom_package_info_fields(&self) -> Vec<String> {
+        vec![PUBLISHED_AT_MANIFEST_FIELD.to_string()]
+    }
+
+    fn custom_dependency_key(&self) -> Option<String> {
+        None
+    }
+
+    fn resolve_custom_dependency(
+        &self,
+        _dep_name: move_symbol_pool::Symbol,
+        _info: &CustomDepInfo,
+    ) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+
+pub fn resolve_manifest_lookup(
```

Comment:
This name is a little confusing to me -- what is it resolving, what are we looking up?  How about `manifest_custom_properties`?